Some changes according to a review by askask

dankwart-de opened the pull request:
- Code example in README doesn't work
- Lock not released in PbsJobManager.runJob
- Failed PBS job results in an NPE instead of an error
- Infinite loop in BEJob

This did not need to be fixed:
- Gradle build fails

askask commented:
- I'm not sure about the changes regarding the working directory, does that mean you can set the log directory only per manager?
- Did you forget the fix for #9?

Otherwise it looks good

dankwart-de commented:
No, you can set the logging directory per job. However, if you do net set it per job, from where do we get the default dir? For me, the job manager is the class to use. However, default directory handling introduces some new questions.

Sorry, nope did not forget #9 but I stashed it... I fixed it and added additional try/finally blocks.

askask commented:
- It's okay for me to set the logging per job, I just missed the method.
- Looks good to me now, thanks
